packages/client/verifiable-credentials/src/presentation/getNfts.ts

```diff
@@ -3,40 +3,59 @@ import { Nft, isVcChain } from "@/verifiableCredentialsSDK";
 import { crossmintAPI } from "../crossmintAPI";
 import { CrossmintWalletNft } from "../types/nfts";
 
-export async function getWalletNfts(chain: string, wallet: string) {
+async function* fetchPaginatedData(url: string): AsyncGenerator<any[]> {
```

Comment:
Oh sorry I should have been more clear! But everything you did here is directly usable. Only one follow up:

- instead of yielding the whole page let's yield one by one

```typescript
              for (const item of data) {
                    yield item;
                }

                if (data.length < perPage) {
                    hasMore = false;
                } else {
                    console.debug(`Processed ${data.length} items from page ${page}`);
                    page++;
                }
```

The user facing functions can return the generator. This way:
- We provide the client a generator that gets the best latency possible in the worst case scenario
- They can retrieve only n items and it won't get stuck if the user has like 1 million credentials
- we abstract away the pagination in a single function and the others (getWalletNfts and getWalletsVCNfts) are completely agnostic

Everything else lgtm !